allennlp/common/from_params.py

```diff
@@ -288,3 +293,53 @@ def from_params(cls: Type[T], params: Params, **extras) -> T:
                 kwargs = create_kwargs(cls, params, **extras)
 
             return cls(**kwargs)  # type: ignore
+
+
+def _load_pretrained_module(cls: Type[T], params: Params) -> T:
```

Comment:
the way things are now (with no caching) I feel like this should just be a public method on `Archive`:

```
def get_module(self, path: str, freeze: bool = True):
    ...
```

and then the code path here just calls that using the params?

That accomplishes two things:

(1) it gives us a way to document this functionality, and
(2) it allows people to use this even if they're not using our `Params` setup.

thoughts?

Reply:
you can still do the same `isinstance` check you were doing before, just do it inside `create_kwargs`

Reply:
that's what i meant by "I can shift type checking out of `get_module` in `from_params`". Since `get_module` is public and doesn't do type check, user can load arbitrary thing. The one loaded from `from_params` would yet be type checked.

Reply:
.. i am fine with it btw, was just pointing it out.

Reply:
@matt-gardner, I am assuming that you don't have any objection here and updating the code. Please let me know if it's otherwise.

Reply:
@joelgrus, archive is a named tuple. I'm making a public function in archival.py as:
```
def load_archive_module(archive_file: str, module_path: str, freeze: bool = True) -> Module:
      ...
```
works? We can later implement caching in archival.py with `cache: Dict[str, Archive]`. Not in this pr though.

Reply:
you can add methods to NamedTuples, just add the method to `Archive` and call it `extract_module` or something like that

Reply:
sounds good, didn't know that.

Reply:
@joelgrus  Would it be better to have `extract_module` on the model than archive? Seems more natural place to me.